staging/src/k8s.io/apiextensions-apiserver/test/integration/table_test.go

```diff
@@ -249,11 +271,23 @@ func TestTableGet(t *testing.T) {
 				if got, expected := tbl.Rows[0].Cells[5], "[1,2,3]"; got != expected {
 					t.Errorf("expected cell[5] to equal %q, got %q", expected, got)
 				}
-				// Validate extra column for v1
+				// Validate extra columns for v1
 				if i == 1 {
 					if got, expected := tbl.Rows[0].Cells[6], int64(5); got != expected {
 						t.Errorf("expected cell[6] to equal %q, got %q", expected, got)
 					}
+					if got, expected := tbl.Rows[0].Cells[7], "[1,3,5,7]"; got != expected {
```

Comment:
Is this the output most users would expect for a string array?  It fits JSON - but does it match what most users would think an array would look like?

In general i’d expect to see more examples here in the tests - what does a mixed array of numbers and strings look like, what does an array of nested maps look like, what does an array with null, empty strings, and other values look like.

If the current test doesn’t work well we may need to restructure it as a table driven test where the inputs are a column definition and an object and the output is a cell (which we call deep equal on).

We only get one chance to change this - once we release this becomes part of our api contract forever (this is a v1 api to a CRD) so we don’t want to leave any corner cases undescribed or undiscussed.  We’re effectively “using up” our ability to make format=string special so we need to make the best use of it we can.

Reply:
>Is this the output most users would expect for a string array? It fits JSON - but does it match what most users would think an array would look like?

I originally experimented with other forms of output here, but ended up settling on this as it matched what we were already returning when the match was a slice. I think it makes sense to return the same output whenever it is a slice, whether that slice represents a single leaf, or aggregated results from multiple leaves.

> We only get one chance to change this - once we release this becomes part of our api contract forever (this is a v1 api to a CRD) so we don’t want to leave any corner cases undescribed or undiscussed. We’re effectively “using up” our ability to make format=string special so we need to make the best use of it we can.

I completely agree with this. My concern is that we already effectively made the decision when we determined how a slice would be rendered if it represented a single leaf. Given that that decision has already been made, I think it makes sense to be consistent with that.

> In general i’d expect to see more examples here in the tests - what does a mixed array of numbers and strings look like, what does an array of nested maps look like, what does an array with null, empty strings, and other values look like.

> If the current test doesn’t work well we may need to restructure it as a table driven test where the inputs are a column definition and an object and the output is a cell (which we call deep equal on).

This makes sense, I'll work on restructuring the tests to make it simpler to add more test cases like this.